core/schema/host.graphqls

```diff
@@ -6,32 +6,19 @@ extend type Query {
 "Information about the host execution environment"
 type Host {
```

Comment:
Side question: how will `Host` even work when we have a long-lived remote API server? Per option 2 in https://github.com/dagger/dagger/issues/3347 it'll live in the OCI image, which might even be in a separate VM/machine, up to two "hosts" away from the client. With multiple clients, what does `workdir` even mean? How are client-side env vars accessed through the API?

Do we need to get started on `Session` or something that holds exposed env/secrets + filesync'd dirs?

Reply:
> Side question: how will `Host` even work when we have a long-lived remote API server? Per option 2 in #3347 it'll live in the OCI image, which might even be in a separate VM/machine, up to two "hosts" away from the client.

I assumed the plan was to implement it on top of `llb.Local`. Once the engine is fully remote, that would require exposing some of the underlying buildkit API (or something equivalent) to support filestreaming etc.

> With multiple clients, what does `workdir` even mean?

Same as "context" in docker build. Each client has its own workdir, and streams its content. Under the hood it's all `llb.Local`.

 > How are client-side env vars accessed through the API?

I'm guessing some sort of callback? Once engine is remote, I guess we could piggyback on secret callbacks?

Note: I think this point is orthogonal to my proposal, as it applies equally to the current API no?

> Do we need to get started on `Session` or something that holds exposed env/secrets + filesync'd dirs?

I think a regular buildkit session handles it just fine, but I might be missing something.

Reply:
> Note: I think this point is orthogonal to my proposal, as it applies equally to the current API no?

Agree, I wanted to raise it because it started to feel like we were rearranging deck chairs on the Titanic. :smile: 

> I think a regular buildkit session handles it just fine, but I might be missing something.

I think what's throwing me off is there's currently only one session and it wraps the entire engine API. I guess we'll be changing that?

https://github.com/dagger/dagger/blob/76e4482adbe9b32fe767a0b4e2b79968422106cc/engine/engine.go#L110-L118

Related to my previous comment. Maybe I'm just getting caught up. :)